src/main/java/uk/gov/hmcts/reform/blobrouter/services/BlobVerifier.java

```diff
@@ -35,20 +37,38 @@ public BlobVerifier(
         }
     }
 
-    public boolean verifyZip(String blobName, byte[] rawBlob) {
+    public VerificationResult verifyZip(String blobName, byte[] rawBlob) {
         try (var zis = new ZipInputStream(new ByteArrayInputStream(rawBlob))) {
 
             ZipVerifiers.verifyZip(zis, publicKey);
-            return true;
+            return ok();
         } catch (DocSignatureFailureException ex) {
             logger.info("Invalid signature. Blob name: {}", blobName, ex);
-            return false;
+            return error("Invalid signature");
         } catch (InvalidZipArchiveException ex) {
             logger.info("Invalid zip archive. Blob name: {}", blobName, ex);
-            return false;
+            return error("Invalid zip archive");
         } catch (IOException ex) {
             logger.info("Error occurred when verifying file. Blob name: {}", blobName, ex);
-            return false;
+            return error(null);
```

Comment:
Do we distinguish between errors with message and errors with null? This last one is some generic failure.

Reply:
How about `"Error loading file"`?

Reply:
I wonder if under this IOException there might be a recoverable failure, not strictly related with the file. Shouldn't we treat it as Exception?

Reply:
done in new PR: https://github.com/hmcts/blob-router-service/pull/279